src/software/new_geom/segment.h

```diff
@@ -89,9 +90,10 @@ class Segment final
         {
             // If the segment and point are in a perfect vertical line, we must use Y
             // coordinate centric logic
-            if ((std::abs(point.x() - getEnd().x()) < GeomConstants::FIXED_EPSILON) &&
-                (std::abs(getEnd().x() - getSegStart().x()) <
-                 GeomConstants::FIXED_EPSILON))
+            if (almostEqual(point.x(), getEnd().x(), GeomConstants::FIXED_EPSILON,
+                            GeomConstants::ULPS_EPSILON_TEN) &&
+                almostEqual(getEnd().x(), getSegStart().x(), GeomConstants::FIXED_EPSILON,
+                            GeomConstants::ULPS_EPSILON_TEN))
```

Comment:
Hmm, the epsilon to use should be decided on by the situation it's used in. Should we expose this as an argument for `colinear`? Then it really should be propogated up to a point where we have some intuition for what reasonable epsilon values would be?

Reply:
Ah I see what you mean, maybe introduce a default parameter EPSILON? so for situations where we need to loosen it, we add an extra EPSILON argument?

Reply:
Yes, IMO that parameter should be present on pretty much all geom functions, but in the interests of keeping this in the scope of this PR, maybe let's just add it to any functions we modified?

Reply:
Added extra arguments with defaults to geom util functions